server/src/main/java/org/elasticsearch/ingest/IngestStats.java

```diff
@@ -76,6 +81,9 @@ public static IngestStats read(StreamInput in) throws IOException {
                 var processorName = in.readString();
                 var processorType = in.readString();
                 var processorStat = readStats(in);
+                // pass these name and type through the local names and types cache to canonical-ize them
+                processorName = namesAndTypesCache.computeIfAbsent(processorName, k -> k);
+                processorType = namesAndTypesCache.computeIfAbsent(processorType, k -> k);
```

Comment:
Just a thought, I have no benchmarked this: I wonder, `processorType` is a very very limited set of values isn't it? Maybe this could use a static `StringLiteralDeduplicator`?

Reply:
EHHHHHHHHHHHHHHHHHHHHH, I'd want to benchmark it. I agree that we could save the space statically (there are only N ingest processors, and they're type strings are relatively fixed at Elasticsearch boot time -- barring plugins and the like), but then we'd be eating the concurrency overhead of multiple threads potentially banging on the same map sometimes.

I was kindof enamored with this being a local on-the-stack reference to a short lived data structure and not having to worry about that. BUT I'll write it as a task for the follow up task/issue that I mentioned in the other comment.

Reply:
I benchmarked this a while back and the whole thing is essentially just a function of whether or not you have enough distinct strings here to create churn on the internal map of the deduplicator.
If there's no writes, a CHM and a normal hash map are remarkably close in read performance these days.
No need to make changes here now, but I think it's safe to use the deduplicator **if** we know that we won't have all that many writes. It feels like that would be true here because we're probably retrieving statistics across the same set of pipelines/processors orders of magnitude more frequently than we update them but I might not see all corner cases here :)

Reply:
That said, if we just change the transport format, who cares :P